### PR TITLE
feat: validate result returns boolean and collection results

### PR DESCRIPTION
**Pull request by SamAg19**

### CODE REVIEW

**Review by rajkharvar — APPROVED** [review]

File: contracts/Staking.sol [hunk 1]
```diff
@@ -19,13 +19,13 @@ contract Staking is AccessControlEnumerable {
 
     function setPermission(
         address sender
-    ) external onlyRole(STAKING_ADMIN_ROLE) {
+    ) external onlyRole(DEFAULT_ADMIN_ROLE) {
```
Comment (hrishikeshio) [inline contracts/Staking.sol:22]: why change this? unrelated change.

**Review by hrishikeshio — CHANGES_REQUESTED** [review]

File: contracts/Staking.sol (hunk 1 above)
Comment (SamAg19) [reply contracts/Staking.sol:22]: Unrelated to the issue created but a change is required to do since only the DEFAULT_ADMIN_ROLE can grant/revoke roles so using `STAKING_ADMIN_ROLE` won't be feasible. @hrishikeshio

Comment (SamAg19) [reply contracts/Staking.sol:22]: Remove `setPermission` and `revokePermission`

**Review by hrishikeshio — APPROVED** [review]

**Review by rajkharvar — APPROVED** [review]

**Review by SkandaBhat — APPROVED** [review]
